src/mapreduce.jl

```diff
@@ -160,25 +160,76 @@ end
 ## reducedim ##
 ###############
 
-@inline reducedim(op, a::StaticArray, ::Val{D}) where {D} = mapreducedim(identity, op, a, Val{D})
-@inline reducedim(op, a::StaticArray, ::Val{D}, v0) where {D} = mapreducedim(identity, op, a, Val{D}, v0)
+@inline reducedim(op, a::StaticArray, ::Type{Val{D}}) where {D} = mapreducedim(identity, op, a, Val{D})
```

Comment:
FYI: This possibly wasn't a typo. I had been thinking about [this PR](https://github.com/JuliaLang/julia/pull/22475) for a long time :)

However, we should probably add a `if VERSION < v"0.7"` or something (or simply support both?)